uportal-war/src/main/java/org/jasig/portal/permission/target/EntityTargetProviderImpl.java

```diff
@@ -126,61 +167,24 @@ public IPermissionTarget getTarget(String key) {
          * Attempt to find matching entities for each allowed entity type.
          * Any matching entities will be added to our collection.
          */
-        
-        for (String type : allowedEntityTypes) {
-            Set<JsonEntityBean> entities = groupListHelper.search(type, term);
+        for (TargetType targetType : allowedTargetTypes) {
+            Set<JsonEntityBean> entities = groupListHelper.search(targetType.toString(), term);
             for (JsonEntityBean entity : entities) {
-                IPermissionTarget target = new PermissionTargetImpl(entity.getId(), entity.getName());
+                IPermissionTarget target = new PermissionTargetImpl(entity.getId(), entity.getName(), targetType);
                 matching.add(target);
             }
         }
 
         if (IPermission.ALL_CATEGORIES_TARGET.contains(term)) {
-            matching.add(getAllCategoriesTarget());
+            matching.add(ALL_CATEGORIES_TARGET);
         } else if (IPermission.ALL_PORTLETS_TARGET.contains(term)) {
-            matching.add(getAllPortletsTarget());
+            matching.add(ALL_PORTLETS_TARGET);
         } else if (IPermission.ALL_GROUPS_TARGET.contains(term)) {
-            matching.add(getAllGroupsTarget());
+            matching.add(ALL_GROUPS_TARGET);
         }
 
         // return the list of matching targets
         return matching;
     }
 
-    /**
-     * Return an IPermissionTarget matching all categories.  This target will
-     * represent any generic category-type entity, allowing administrators
-     * to set permissions on ungrouped entity items.
-     * 
-     * @return all categories target
-     */
-    protected IPermissionTarget getAllCategoriesTarget() {
```

Comment:
API change.  Removes a utility method previously exposed to sub-classes, which are possible for this non-final class.  New static members for these targets are `private` rather than `protected` so not visible to sub-classes so the update to use the new API is less trivial.  Worth keeping these around `@deprecated` for a release or two? Their implementations could be trivial, of course, returning the new static members.